src/Core/Components/Checkbox/FluentCheckbox.razor.cs

```diff
@@ -0,0 +1,99 @@
+// ------------------------------------------------------------------------
+// MIT License - Copyright (c) Microsoft Corporation. All rights reserved.
+// ------------------------------------------------------------------------
+
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection.Metadata;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// The FluentCheckbox component is used to render a checkbox input.
+/// </summary>
+public partial class FluentCheckbox : FluentInputImmediateBase<bool?>, IFluentComponentElementBase
+{
+    private const string JAVASCRIPT_FILE = FluentJSModule.JAVASCRIPT_ROOT + "TextInput/FluentTextInput.razor.js";
+
+    /// <inheritdoc />
+    [Parameter]
+    public ElementReference Element { get; set; }
+
+    /// <summary>
+    /// Gets or sets the checked state of the checkbox
+    /// </summary>
+    [Parameter]
+    public bool? Checked { get; set; }
```

Comment:
Why do we need this property ? I think only `Value` must be used.

Reply:
This is a discussion that I would like to get : 
About the indeterminate value, can we suppose to go to this solution. 

Value : 
- False : Unchecked
- Null : Indeterminate
- True Checked

Ok for you ?

Reply:
Yes and we need to have a `bool ThreeState` property to specify if the user can switch between the 3 states: unchecked, checked, indeterminate (using this order). See the existing demo site.